docs/api/useInvalidator.md

```diff
@@ -0,0 +1,50 @@
+---
+title: useInvalidator()
+---
+
+```typescript
+function useInvalidator<Params extends Readonly<object>, S extends Schema>(
+  selectShape: ReadShape<S, Params, any>,
+): (params: Params) => void;
+```
+
+Mostly useful for imperatively invalidating the cache, with a similar signiature to
+[useFetcher](./useFetcher). When used in conjunction with `invalidIfStale` it can force a
+component to re-suspend even if it has already fetched the data.
+
+## Example
+
+```typescript
+import { Resource, RequestOptions } from 'rest-hooks';
+
+export default class ArticleResource extends Resource {
+  readonly id: string = null;
+  readonly title: string = '';
+  // ...
+
+  /** Used as default options for every RequestShape */
+  static getRequestOptions(): RequestOptions {
+    return {
+      invalidIfStale: true,
+    };
+  }
+}
+```
+
+```tsx
+function ArticleName({ id }: { id: string }) {
+  const asset = useResource(ArticleResource.singleRequest(), { id });
+  const invalidate = useInvalidator(ArticleResource.singleRequest());
```

Comment:
nit: a better example might be creating a new hook that invalidates on unmount that wraps both useInvalidator and useEffect(). This shows people good hook patterns.